core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -187,10 +286,25 @@ public TermSuggestionBuilder minWordLength(int minWordLength) {
      * value higher than 1 is specified then the number cannot be fractional.
      */
     public TermSuggestionBuilder minDocFreq(float minDocFreq) {
+        if (minDocFreq < 0.0f) {
```

Comment:
Can you put a few tests for illegal values (one for each setter that has arg-checks is enough I think) into TermSuggestionBuilderTests?